File: ABSample-ObjC/ABSample-ObjC/ViewController.m
//
//  ViewController.m
//  ABSample-ObjC
//
//  Created by Stas Kochkin on 21/06/2018.
//  Copyright © 2018 Appodeal. All rights reserved.
//

#import "ViewController.h"
#import "AdMediation.h"


@interface ViewController ()

@end

@implementation ViewController

- (IBAction)presentBanner {
    [self showAd:AdTypeBanner];
}

- (IBAction)presentRewardedVideo {
    [self showAd:AdTypeRewardedVideo];
}

- (IBAction)presentInterstitial:(id)sender {
    [self showAd:AdTypeInterstitial];
}

- (void)showAd:(AdType)adType {
    if ([AdMediation.sharedInstance.provider hasReadyAd:adType]) {
        [AdMediation.sharedInstance.provider presentAd:adType rootViewController:self];
    }
}

@end

File: ABSample-Swift/ABSample-Swift/ViewController.swift
//
//  ViewController.swift
//  ABSample-Swift
//
//  Created by Stas Kochkin on 21/06/2018.
//  Copyright © 2018 Appodeal. All rights reserved.
//

import UIKit

private extension String {
    var adType : AdType? {
        get {
            var adType : AdType?
            switch self {
            case "BannerCell":
                adType = .banner
            case "RewardedVideoCell":
                adType = .rewarded
            case "InterstitialCell":
                adType = .rewarded
            default:
                break
            }
            return adType
        }
    }
}

class ViewController: UITableViewController {
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        let cell = tableView.cellForRow(at: indexPath)!
        
        guard let adType = cell.reuseIdentifier?.adType else {
            return
        }
        
        if AdMediation.sharedInstance.provider?.hasReadyAd(adType) ?? false {
            AdMediation.sharedInstance.provider!.presentAd(adType, self)
        }
    }
}

